stdlib/public/core/Unicode.swift

```diff
@@ -71,51 +73,131 @@ public protocol UnicodeCodecType {
 
 /// A codec for [UTF-8](http://www.unicode.org/glossary/#UTF_8).
 public struct UTF8 : UnicodeCodecType {
+  // See Unicode 8.0.0, Ch 3.9, UTF-8.
+  // http://www.unicode.org/versions/Unicode8.0.0/ch03.pdf
```

Comment:
Personally it would have helped me when reading the original code, but maybe that's just me ;).